test/controllers/users/registrations_controller_test.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Users
+  class RegistrationsControllerTest < ActionDispatch::IntegrationTest
+    test "should register new user" do
+      user = {
+        name: "name",
+        surname: "surname",
+        email: "test@example.com",
+        password: "password",
+        password_confirmation: "password"
+      }
+
+      post "/user", params: {
+        user: user
+      }
+
+      user_list = User.find_by!(name: user[:name], surname: user[:surname], email: user[:email])
+      assert user_list
```

Comment:
Also you can use `assert_changes` or `assert_difference` here

```ruby
assert_changes -> { User.count } do
  post "/user", params: { user: user }
end
```

https://api.rubyonrails.org/v7.0.3.1/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_changes

Write a version with `assert_difference` yourself :)

https://api.rubyonrails.org/v7.0.3.1/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_difference